code/edgemedia/src/main/java/com/adobe/marketing/mobile/edge/media/internal/MediaXDMEventHelper.kt

```diff
@@ -74,39 +74,37 @@ internal class MediaXDMEventHelper {
             sessionDetails.hasResume = forceResume || mediaInfo.isResumed // To also handle the internally triggered resume by the SDK for long running sessions >= 24 hours
 
             metadata.forEach { (key, value) ->
-                run {
-                    if (!standardMediaMetadataSet.contains(key)) {
-                        return@forEach
-                    }
-
-                    when (key) {
-                        // Video standard metadata cases
-                        MediaConstants.VideoMetadataKeys.AD_LOAD -> sessionDetails.adLoad = value
-                        MediaConstants.VideoMetadataKeys.ASSET_ID -> sessionDetails.assetID = value
-                        MediaConstants.VideoMetadataKeys.AUTHORIZED -> sessionDetails.authorized = value
-                        MediaConstants.VideoMetadataKeys.DAY_PART -> sessionDetails.dayPart = value
-                        MediaConstants.VideoMetadataKeys.EPISODE -> sessionDetails.episode = value
-                        MediaConstants.VideoMetadataKeys.FEED -> sessionDetails.feed = value
-                        MediaConstants.VideoMetadataKeys.FIRST_AIR_DATE -> sessionDetails.firstAirDate = value
-                        MediaConstants.VideoMetadataKeys.FIRST_DIGITAL_DATE -> sessionDetails.firstDigitalDate = value
-                        MediaConstants.VideoMetadataKeys.GENRE -> sessionDetails.genre = value
-                        MediaConstants.VideoMetadataKeys.MVPD -> sessionDetails.mvpd = value
-                        MediaConstants.VideoMetadataKeys.NETWORK -> sessionDetails.network = value
-                        MediaConstants.VideoMetadataKeys.ORIGINATOR -> sessionDetails.originator = value
-                        MediaConstants.VideoMetadataKeys.RATING -> sessionDetails.rating = value
-                        MediaConstants.VideoMetadataKeys.SEASON -> sessionDetails.season = value
-                        MediaConstants.VideoMetadataKeys.SHOW -> sessionDetails.show = value
-                        MediaConstants.VideoMetadataKeys.SHOW_TYPE -> sessionDetails.showType = value
-                        MediaConstants.VideoMetadataKeys.STREAM_FORMAT -> sessionDetails.streamFormat = value
-
-                        // Audio standard metadata cases
-                        MediaConstants.AudioMetadataKeys.ALBUM -> sessionDetails.album = value
-                        MediaConstants.AudioMetadataKeys.ARTIST -> sessionDetails.artist = value
-                        MediaConstants.AudioMetadataKeys.AUTHOR -> sessionDetails.author = value
-                        MediaConstants.AudioMetadataKeys.LABEL -> sessionDetails.label = value
-                        MediaConstants.AudioMetadataKeys.PUBLISHER -> sessionDetails.publisher = value
-                        MediaConstants.AudioMetadataKeys.STATION -> sessionDetails.station = value
-                    }
+                if (!standardMediaMetadataSet.contains(key)) {
```

Comment:
Since `when` will only assign the value for the keys defined in the `when` block, is this check needed?